js/Wizard.js

```diff
@@ -170,12 +170,14 @@ function checkSeed(seed) {
 }
 
 function restoreWalletCheckViewSpendAddress(walletmanager, nettype, viewkey, spendkey, addressline){
-    var addressOK = (viewkey.length > 0 || spendkey.length > 0) ? walletmanager.addressValid(addressline, nettype) : false
-    var viewKeyOK = (viewkey.length > 0) ? walletmanager.keyValid(viewkey, addressline, true, nettype) : true
-    // Spendkey is optional
-    var spendKeyOK = (spendkey.length > 0) ? walletmanager.keyValid(spendkey, addressline, false, nettype) : true
-
-    return addressOK && viewKeyOK && spendKeyOK
+    var results = [];
+    // addressOK
+    results[0] = (viewkey.length > 0 || spendkey.length > 0) ? walletmanager.addressValid(addressline, nettype) : false
```

Comment:
```suggestion
    results[0] = walletmanager.addressValid(addressline, nettype);
```

Same for other couple checks

Reply:
Good call. Fixed, thank you.